clang/lib/Driver/Driver.cpp

```diff
@@ -3903,8 +3990,11 @@ class OffloadingActionBuilder final {
       WrapDeviceOnlyBinary = Args.hasArg(options::OPT_fsycl_link_EQ);
       auto *DeviceCodeSplitArg =
           Args.getLastArg(options::OPT_fsycl_device_code_split_EQ);
-      DeviceCodeSplit = DeviceCodeSplitArg &&
-                        DeviceCodeSplitArg->getValue() != StringRef("off");
+      bool SplitOff = DeviceCodeSplitArg &&
+                      DeviceCodeSplitArg->getValue() == StringRef("off");
+      DeviceCodeSplit =
+          !SplitOff && (DeviceCodeSplitArg ||
+                        Args.getLastArg(options::OPT_fsycl_device_code_split));
```

Comment:
`OPT_fsycl_device_code_split` is an alias to `OPT_fsycl_device_code_split_EQ` so you only have to check if `OPT_fsycl_device_code_split_EQ` arg is set && !off (similar to what you did below)

Reply:
ok